src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -277,31 +277,16 @@ public CatalogObjectMetadata updateObjectMetadata(String bucketName, String name
             throw new TagsIsNotValidException(tags.get(), "tags");
         }
 
-        //synchronize project name values
-        for (KeyValueLabelMetadataEntity metadata : catalogObjectRevisionEntity.getKeyValueMetadataList()) {
-            if (metadata.getKey().equals(PROJECT_NAME)) {
-                metadata.setValue(projectName.orElse(""));
-            }
+        if (!catalogObjectRevisionEntity.getProjectName().equals(projectName.orElse("")) ||
+            !catalogObjectRevisionEntity.getTags().equals(tags.orElse(""))) {
+            catalogObjectRevisionEntity = buildCatalogObjectRevisionEntity("The project name or/and tags metadata are updated",
```

Comment:
better call this catalogObjectNewRevisionEntity

Reply:
The variable assignment is useless. I removed it